`plat/xen`: Building with Clang fails on ARM64: `incompatible pointer to integer conversion assigning`

### Describe the bug

Building applications with Clang for Xen ARM64 results in a build error.

### Steps to reproduce

Build C Hello with Clang for Xen/ARM64 in the [`catalog-core` repository](https://github.com/unikraft/catalog-core). Follow the steps:

1. Clone the repository:

   ```console
   git clone https://github.com/unikraft/catalog-core
   ```

1. Enter the `c-hello/` directory:

   ```console
   cd catalog-core/c-hello/
   ```
   
1. Switch to the `scripts` branch:

   ```console
   git checkout scripts
   ```

1. Build C Hello for Xen/ARM64:

   ```console
   ./build.xen.arm64
   ```

   The build error occurs.

### Expected behavior

Build is successful, without errors.

### Which architectures were you using or does this bug affect?

arm64

### Which platforms were you using or does this bug affect?

xen

### Relevant log output

```shell
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/plat/xen/arm/setup64.c:176:14: error: incompatible pointer to integer conversion assigning to '__u64' (aka 'unsigned long') from 'char[
8]' [-Wint-conversion]                                                                                                                                                                        
  176 |         bi->cmdline = CONFIG_UK_NAME;                                                                                                                                                 
      |                     ^ ~~~~~~~~~~~~~~                                                                                                                                                  
/home/razvan/orgs/unikraft/catalog-core/repos/unikraft/plat/xen/arm/setup64.c:184:16: error: incompatible pointer to integer conversion assigning to '__u64' (aka 'unsigned long') from 'const
 char *' [-Wint-conversion]                                                                                                                                                                   
  184 |                         bi->cmdline = fdtcmdline;
```

--- a/plat/xen/arm/setup64.c
+++ b/plat/xen/arm/setup64.c
@@ -173,7 +173,7 @@ static inline void _get_cmdline(struct ukplat_bootinfo *bi)
 
 	UK_ASSERT(bi);
 
-	bi->cmdline = CONFIG_UK_NAME;
+	bi->cmdline = (__uptr)CONFIG_UK_NAME;
 	bi->cmdline_len = sizeof(CONFIG_UK_NAME) - 1;
 
 	fdtchosen = fdt_path_offset(HYPERVISOR_dtb, "/chosen");
@@ -181,7 +181,7 @@ static inline void _get_cmdline(struct ukplat_bootinfo *bi)
 		fdtcmdline = fdt_getprop(HYPERVISOR_dtb, fdtchosen, "bootargs",
 					 &len);
 		if (fdtcmdline) {
-			bi->cmdline = fdtcmdline;
+			bi->cmdline = (__uptr)fdtcmdline;
 			bi->cmdline_len = len;
 		}
 	}

